packages/compass-crud/src/stores/crud-store.js

```diff
@@ -626,7 +626,8 @@ const configureStore = (options = {}) => {
         projection,
         collation,
         maxTimeMS,
-        promoteValues: false
+        promoteValues: false,
+        bsonRegExp: true
```

Comment:
I'm like 99.9% sure that this is the behavior we want because without this flag we never get `BSONRegExp` instances back from the driver and code in document view / hadron-document / hadorn-type-checker is written as if this should be `BSONRegExp` and not just RegExp, but please tell me if you see any issues with that